conf/script/pre_disruptive_upgrade.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+set -eu -o pipefail
+
+BACKUP_SOURCE_DIR=/etc/cortx/ha
+BACKUP_DEST_DIR=/opt/seagate/cortx/ha_conf_backup
+consul=/usr/bin/consul
+HARE_DIR=/var/lib/hare
+
+backup_consul(){
+
+    echo "Taking the backup of consul"
+    consul_kv_dump=$HARE_DIR/consul-kv-dump.json
+    if [[ -f $consul_kv_dump ]]; then
+        cmd < $consul_kv_dump | xz > ${consul_kv_dump%.json}-$(date '+%Y%m%d%H%M%S').json.xz
+    fi
+    $consul kv export > $consul_kv_dump
```

Comment:
What if export fails?

Reply:
I am still trying to understand this consul backup code. I got this from HARE team. I will revert back on this.

Reply:
So, this code works like following way:
1. If the consul dump is already there, take the data in compressed format
2. Then export latest version consul KV.

So, if this fails, I have added code to exit because if upgrade fails in between, we will not be able to retrieve the consul data. Please let me know your thoughts.